Cython/Compiler/Symtab.py

```diff
@@ -724,10 +726,15 @@ def declare_var(self, name, type, pos,
                 cname = name
             else:
                 cname = self.mangle(Naming.var_prefix, name)
-        if type.is_cpp_class and visibility != 'extern':
-            type.check_nullary_constructor(pos)
         entry = self.declare(name, cname, type, pos, visibility)
         entry.is_variable = 1
+        if type.is_cpp_class and visibility != 'extern':
+            if self.directives['cpp_locals']:
+                entry.is_cpp_optional = True
+                self.use_utility_code(
+                        Code.UtilityCode.load_cached("OptionalLocals", "CppSupport.cpp"))
```

Comment:
The `Entry` could have a method for this (`make_cpp_optional()`), which would actually store the utility code in `Entry.utility_code` and not use it globally.